src/Run.js

```diff
@@ -16,9 +16,23 @@ function Run(ref, json_q, callback,progressObj) {
       var image = drawarray[pos].image;
       var i = drawarray[pos].i;
       var input = ref.images[image].steps[i - 1].output;
-      ref.images[image].steps[i].draw(ref.copy(input), function onEachStep() {
-        drawStep(drawarray, ++pos);
-      },progressObj);
+      var outputs_arr = [];
+      for (let opt in ref.images[image].steps) {
+        if (opt >= i) break;
+        outputs_arr.push(ref.images[image].steps[opt].output);
+      }
+      input.history = ref.copy(outputs_arr);
+      input.getStep = function(offset) {
```

Comment:
Let's break this out into a separate function definition if we can? And name it with `function getStep(` so it shows up as a named function in debugging.

Reply:
Oh, Nice observation @jywarren . Doing this now!!